addons/GME/Scripts/GameCode/GME/Character/SCR_ChimeraCharacter.c

```diff
@@ -1,23 +1,47 @@
-
-
+//------------------------------------------------------------------------------------------------
 modded class SCR_ChimeraCharacter : ChimeraCharacter
 {
 	[RplProp(onRplName: "GME_OnVisibilityValueUpdated")]
 	protected bool m_bGME_isVisible = true;
 	
-	void GME_SetVisibility(bool visible)
-	{		
-		m_bGME_isVisible = visible;
-		Replication.BumpMe();
-		this.GME_OnVisibilityValueUpdated();
-	}
+	//------------------------------------------------------------------------------------------------
+	//! Apply global AI skill
+	override void EOnInit(IEntity owner)
+	{
+		super.EOnInit(owner);
+		
+		if (!GetGame().InPlayMode() || !Replication.IsServer())
+			return;
+		
+		SCR_BaseGameMode gameMode = SCR_BaseGameMode.Cast(GetGame().GetGameMode());
+		if (!gameMode)
+			return;
+		
+		SCR_AICombatComponent combatComponent = SCR_AICombatComponent.Cast(FindComponent(SCR_AICombatComponent));
+		if (!combatComponent)
+			return;
+		
+		EAISkill skill = gameMode.GME_GetGlobalAISkill();
+		if (skill == EAISkill.NONE)
+			return;
 	
+		combatComponent.SetAISkill(skill);
+	}
 	
+	//------------------------------------------------------------------------------------------------
 	void GME_OnVisibilityValueUpdated()
 	{
 		if (m_bGME_isVisible)
 			this.SetFlags(EntityFlags.VISIBLE|EntityFlags.TRACEABLE, m_bGME_isVisible);
 		else
 			this.ClearFlags(EntityFlags.VISIBLE|EntityFlags.TRACEABLE);
 	}
-};
\ No newline at end of file
+	
+	//------------------------------------------------------------------------------------------------
+	void GME_SetVisibility(bool visible)
+	{		
+		m_bGME_isVisible = visible;
+		Replication.BumpMe();
```

Comment:
Is it on purpose the bump happens before the ``onVisibilityValueUpdated()`` function is called locally? As I noticed in some of the other rpl functions we call locally first then bump?

Reply:
ah, gotcha. Explains why it highlighted visibility as new, as I was pretty certain we had that already 😅 

fair. I assumed the order was not important due to the nature of it, but wasn't 100% sure if we wanted to adhere a specific order of it, to ensure the authority is refreshed before the replication bump goes out. Although I guess it happens to quick after each other that it would not matter in any case.